todo-app/src/stores/TodoStore.js

```diff
@@ -0,0 +1,123 @@
+import { action, observable, runInAction } from 'mobx';
+
+const addTodoFormDefault = {
+  description: '',
+  deadline: '',
+}
+
+const API_ROOT = 'https://hyf-react-api.herokuapp.com';
+
+export class TodoStore {
+  @observable
+  todos = []
+
+  @observable
+  addTodoForm = addTodoFormDefault
+
+  @observable
+  todoEditingID = null
+
+  @observable
+  todoEditingText = ''
+
+  //TODOs API
+  @action
+  getTodos = async () => {
+    const res = await fetch(`${API_ROOT}/todos`);
+    const parseResponse = await res.json()
+    runInAction(() => {
+      this.todos = parseResponse
+    })
+  }
+
+  // Change the state of DONE foreach todo
+  @action
+  toggleDone = async todo => {
+    this.todos = this.todos.map((task) => {
+      if (task._id === todo._id) {
+        return {
+          ...task,
+          done: !task.done
+        }
+      }
+      return task
+    })
+    await fetch(`${API_ROOT}/todos/${todo._id}`, {
```

Comment:
What happens if this network request fails? You need add a `catch` and unset the checkbox change made in this case, otherwise you will have incorrect data being displayed to the user.